stl/inc/deque

```diff
@@ -986,19 +986,63 @@ public:
     }
 
     void shrink_to_fit() {
-        size_type _Oldcapacity = _Block_size * _Mapsize();
-        size_type _Newcapacity = _Oldcapacity / 2;
+        if (empty()) {
+            if (_Map() != nullptr) {
+                _Reset_map();
+            }
+            return;
+        }
 
-        if (_Newcapacity < _Block_size * _Minimum_map_size) {
-            _Newcapacity = _Block_size * _Minimum_map_size;
+        const auto _First_block_idx = static_cast<_Map_difference_type>(_Myoff() / _Block_size);
+        const auto _Last_block_idx =
+            static_cast<_Map_difference_type>(((_Myoff() + _Mysize() - 1) / _Block_size + 1) & (_Mapsize() - 1));

```

Comment:
As the calculation is complex here, I think it's better to add comment that `_Last_block_idx` is the past-the-end index.

Reply:
Now I think that "last" possibly have misleading impression, given we're actually operating on a circular buffer. Perhaps we should use another word instead of "last".